libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/RouteProgressObserver.kt

```diff
@@ -1,4 +1,4 @@
-package com.mapbox.navigation.core.trip.session
+package com.mapbox.navigation.base.trip
```

Comment:
Any particular reason this was exposed? The observer is still tied to the `MapboxTripSession`, which is internal. Is there a scenario where the observer would be used without using `core`?

Reply:
Moved it so it lives near the `RouteProgress` and the rest of contracts and models (as it is one of the main observers accessed by most of the modules and needed publicly for clients). Although, that's a great point. Currently `MapboxTripSession` lives in `libnavigation-core` and I can't think of any use case in which you might want to use the observer alone. Can move it back to `libnavigation-core` as part of the missing internal package refactoring https://github.com/mapbox/mapbox-navigation-android/issues/2417 of `libnavigation-core` 👍